ydb/library/login/login.cpp

```diff
@@ -92,6 +92,11 @@ bool TLoginProvider::CheckUserExists(const TString& user) {
     return CheckSubjectExists(user, ESidType::USER);
 }
 
+bool TLoginProvider::CheckSidExistsOrIsNonYdb(const TString& sid) {
+    // non-YDB user's sid format is <login>@<subsystem>
```

Comment:
there are no rules on SID formats. it's intentional.
it could be anything. and even in the login library you could have SIDs with @ symbol.
and it has nothing to do with YDB or any other project that uses this library.

Reply:
> there are no rules on SID formats. it's intentional.

You can't create neither a user or a group with @ in its name

https://github.com/ydb-platform/ydb/blob/d2e4bb83f1c9678aacc2b39ab18c68294ffc1bc4/ydb/library/login/login.cpp#L58

Reply:
okay, I see, but anyway, I think this check looks very weak and I don't think it should have place in the abstract login library. please, keep it abstract, it could be reused anywhere.

Reply:
I've moved check out of the library